src/components/Navigation/components/Item/Item.tsx

```diff
@@ -137,14 +128,38 @@ export class BaseItem extends React.Component<CombinedProps, State> {
       )
     );
 
+    let badgeMarkup: Props['badge'] = null;
```

Comment:
I would just use `React.ReactNode` for the type, it feels weird to use `Props['badge']` since you make it consume the badge prop and transform it